src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseSigner.cs

```diff
@@ -5,23 +5,60 @@
 
 namespace System.Security.Cryptography.Cose
 {
+    /// <summary>
+    /// Provides signing information to be used with sign operations in <see cref="CoseSign1Message"/> and <see cref="CoseMultiSignMessage"/>.
+    /// </summary>
+    /// <seealso cref="CoseSign1Message.SignEmbedded(byte[], CoseSigner, byte[])"/>
+    /// <seealso cref="CoseMultiSignMessage.SignEmbedded(byte[], CoseSigner, CoseHeaderMap?, CoseHeaderMap?, byte[])"/>
+    /// <seealso cref="CoseMultiSignMessage.AddSignatureForEmbedded(CoseSigner, byte[])"/>
     public sealed class CoseSigner
     {
         internal readonly KeyType _keyType;
         internal readonly int? _algHeaderValueToSlip;
         internal CoseHeaderMap? _protectedHeaders;
         internal CoseHeaderMap? _unprotectedHeaders;
+
+        /// <summary>
+        /// Gets the cryptographic key of this instance.
+        /// </summary>
         public AsymmetricAlgorithm Key { get; }
+
+        /// <summary>
+        /// Gets the hash algorithm of this instance.
+        /// </summary>
         public HashAlgorithmName HashAlgorithm { get; }
+
+        /// <summary>
+        /// Gets the signature padding of this instance.
+        /// </summary>
         public RSASignaturePadding? RSASignaturePadding { get; }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CoseSigner"/> class.
+        /// </summary>
+        /// <param name="key">The private key used to sign the content.</param>
```

Comment:
There's no content.  And one day this might be used for counter-signing.  So keep it vague.

```suggestion
        /// <param name="key">The private key to use for signing.</param>
```